signer/signer.go

```diff
@@ -172,3 +190,40 @@ func CheckSignature(csr *x509.CertificateRequest, algo x509.SignatureAlgorithm,
 	}
 	return x509.ErrUnsupportedAlgorithm
 }
+
+// NewSigner generates a new certificate signer from a Root structure
+// If the root structure specifies a force remote, then a remote signer
+// is created, otherwise either a remote or local signer is generated
+// based on the policy. For a local signer, the CertFile and KeyFile
+// need to be defined.
```

Comment:
Maybe a note that this only provides one of the two standard signers (remote/local).